src/components/calcite-label/calcite-label.tsx

```diff
@@ -74,24 +74,72 @@ export class CalciteLabel {
   //--------------------------------------------------------------------------
 
   @Listen("click")
-  onClick({ target }: MouseEvent): void {
-    if (target === this.el || target === this.labelEl || target === this.spanEl) {
-      const forAttr = this.el.getAttribute("for");
-      this.calciteLabelFocus.emit({
-        labelEl: this.el,
-        interactedEl: target as HTMLElement,
-        requestedInput: forAttr
-      });
-      const inputForThisLabel: HTMLElement = forAttr
-        ? document.getElementById(forAttr)
-        : this.el.querySelector("input");
-      if (
-        (inputForThisLabel && inputForThisLabel.nodeName.startsWith("CALCITE-")) ||
-        (inputForThisLabel && inputForThisLabel.nodeName === "INPUT" && target === this.el)
-      ) {
-        inputForThisLabel.click();
+  onClick(event: MouseEvent): void {
+    const target = event.target as HTMLElement;
+    this.calciteLabelFocus.emit({
+      labelEl: this.el,
+      interactedEl: target,
+      requestedInput: this.for
+    });
+
+    // 1. has htmlFor
```

Comment:
It might be nice to extract all of the logic that just returns into a utility function that returns true or false.

Reply:
some function to validate the target and for.